rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/LocalTransferServer.java

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright (c) 2016. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.bookkeeper;
+
+import com.google.common.base.Throwables;
+import com.qubole.rubix.spi.BookKeeperFactory;
+import com.qubole.rubix.spi.CacheConfig;
+import com.qubole.rubix.spi.RetryingBookkeeperClient;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.thrift.shaded.TException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousCloseException;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.SocketChannel;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Created by sakshia on 26/10/16.
+ */
+
+public class LocalTransferServer
+{
+    private static Log log = LogFactory.getLog(LocalTransferServer.class.getName());
+    private static Configuration conf;
+    private static  LocalServer localServer;
+
+    private LocalTransferServer()
+    {
+    }
+
+    public static void main(String[] args)
+    {
+        conf = new Configuration();
+        startServer(conf);
+    }
+
+    public static void startServer(Configuration conf)
+    {
+        localServer = new LocalServer(conf);
+        new Thread(localServer).run();
+    }
+
+    public static void stopServer()
+    {
+        localServer.stop();
+    }
+
+    public static class LocalServer implements Runnable
+    {
+        static ServerSocketChannel listener;
+        Configuration conf;
+
+        public LocalServer(Configuration conf)
+        {
+            this.conf = conf;
+        }
+
+        @Override
+        public void run()
+        {
+            int port = CacheConfig.getLocalServerPort(conf);
+
+            ExecutorService threadPool = Executors.newCachedThreadPool();
+            try {
+                listener = ServerSocketChannel.open();
+                listener.bind(new InetSocketAddress(port));
+                log.info("Listening on port " + port);
+                while (true) {
+                    SocketChannel clientSocket = listener.accept();
```

Comment:
What are the service guarantees for this? Do we know how many connections it can accept? Does it tell when it drops a connection? I hope we have retry in the client code of this, will check when i get there